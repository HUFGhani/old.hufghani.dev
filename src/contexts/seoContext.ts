import { createContext, useContext } from 'react'

interface SEOContextType {
  data: {
    site: {
      siteMetadata: {
        author: string
        url: string
        twitter: string
        titleAlt: string
        siteUrl: string
        siteLanguage: string
        shortName: string
        pathPrefix: string
        description: string
      }
    }
  }
}

export const SEOContext = createContext<SEOContextType | undefined>(undefined)

export const useSEOContext = () => useContext<SEOContextType | undefined>(SEOContext)
