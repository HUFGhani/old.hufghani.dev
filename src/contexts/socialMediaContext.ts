import { createContext, useContext } from 'react'
import SocialMediaInterface from '../interface/socialMediaInterface'

export const SocialMediaContext = createContext<SocialMediaInterface | undefined>(undefined)

export const useSocialMediaContext = () => useContext<SocialMediaInterface | undefined>(SocialMediaContext)
