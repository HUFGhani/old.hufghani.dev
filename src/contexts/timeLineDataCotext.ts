import { createContext, useContext } from 'react'
import TimelineItemInterface from '../interface/timelineItemInterface'

export const TimeLineDataCotext = createContext<TimelineItemInterface | undefined>(undefined)

export const useTimeLineDataCotext = () => useContext<TimelineItemInterface | undefined>(TimeLineDataCotext)
