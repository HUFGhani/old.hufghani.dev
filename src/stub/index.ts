export const SiteMetaData = {
  author: `Hamza Umar Farooq Ghani`,
  url: `https://hufghani.dev`,
  twitter: `@the_ghani`,
  titleAlt: `Hamza Ghani - Personal Website`,
  siteUrl: `https://hufghani.dev`,
  siteLanguage: `en`,
  shortName: `HUFGHANI`,
  pathPrefix: `/`,
  description: `personal Website`,
}
