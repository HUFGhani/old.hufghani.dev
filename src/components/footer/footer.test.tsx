import { render } from '@testing-library/react'
import React from 'react'
import Footer from '.'
import { SEOContext } from '../../contexts/seoContext'
import { SocialMediaContext } from '../../contexts/socialMediaContext'
import { SiteMetaData, SocialMediaData } from '../../stub'

function renderFooter() {
  return render(
    <SEOContext.Provider value={SiteMetaData}>
      <SocialMediaContext.Provider value={SocialMediaData}>
        <Footer />
      </SocialMediaContext.Provider>
    </SEOContext.Provider>
  )
}

describe(`<Footer/>`, () => {
  it(`should render Footer`, () => {
    const { asFragment } = renderFooter()
    expect(asFragment()).toMatchSnapshot()
  })
  it(`should render the copyright text correctly`, () => {
    const { getByTestId } = renderFooter()
    expect(getByTestId(`copyRight-year`)).toHaveTextContent(
      `© 2017 - ${new Date().getFullYear()}, Hamza Umar Farooq Ghani`
    )
  })

  it(`should render the copyright text correctly with aria-labal`, () => {
    const { getByLabelText } = renderFooter()
    expect(getByLabelText(`© 2017 - 2021, Hamza Umar Farooq Ghani`)).toHaveTextContent(
      `© 2017 - ${new Date().getFullYear()}, Hamza Umar Farooq Ghani`
    )
  })
})
