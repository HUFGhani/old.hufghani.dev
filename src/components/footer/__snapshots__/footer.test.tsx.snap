// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render Footer 1`] = `
<DocumentFragment>
  <footer
    class="css-12pe2qd"
  >
    <p
      aria-label="© 2017 - 2021, Hamza Umar Farooq Ghani"
      class="chakra-text css-0"
      data-testid="copyRight-year"
    >
      © 2017 - 2021, Hamza Umar Farooq Ghani
    </p>
    <p
      class="chakra-text css-0"
    >
      Built with Gatsby · Hosted on S3 and CloudFront
    </p>
  </footer>
</DocumentFragment>
`;
